MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -550,7 +553,7 @@ namespace WPEFramework {
                 if ( (g_task_status & TASKS_COMPLETED ) == TASKS_COMPLETED ){
                     if ( (g_task_status & ALL_TASKS_SUCCESS) == ALL_TASKS_SUCCESS ){ // all tasks success
                         LOGINFO("DBG:Maintenance Successfully Completed!!");
-                        m_notify_status=MAINTENANCE_COMPLETE;
+                        notify_status=MAINTENANCE_COMPLETE;
```

Comment:
You can update m_notify_status (class variable) here itself after holding on to status mutex.